lib/shrinkwrap.js

```diff
@@ -33,10 +36,34 @@ function shrinkwrap (args, silent, cb) {
   npm.config.set('production', true)
   readPackageTree(dir, andRecalculateMetadata(iferr(cb, function (tree) {
     var pkginfo = treeToShrinkwrap(tree, !!npm.config.get('dev') || /^dev(elopment)?$/.test(npm.config.get('also')))
-    shrinkwrap_(pkginfo, silent, cb)
+
+    readPackage(iferr(cb, function (data) {
+      chain([
+        [lifecycle, data, 'preshrinkwrap'],
+        [shrinkwrap_, pkginfo, silent],
```

Comment:
I'm suggesting you call a `shrinkwrap` lifecycle either before or after this. Which isn't super important, we just need to document it. For what it's worth, `build` does `preinstall`, _work_, `install`, `postinstall` and`unbuild` runs `preuninstall`, `uninstall`, _work_, `postuninstall`. So seems like _after_ would make more sense, to align with `build`'s behavior.